src/app/features/email-client/email-client.service.ts

```diff
@@ -43,9 +43,22 @@ export class EmailClientService {
       this.getEmailsForEntities(entityList);
 
     if (!recipients.length) {
-      this.alertService.addWarning(
-        $localize`Please fill an email address for this record to use this functionality.`,
-      );
+      await this.dialog
```

Comment:
@Abhinegi2 , now that this is awaiting a dialog response, the test needs to mock the user reaction to close the dialog, otherwise it will never reach the `return false` during the test